### DISCUSSION TITLE
Request from the backend instead of the user's browser?

### DISCUSSION THREAD

**Discussion by OneCyrus** [hoppscotch/hoppscotch#3450] [Q&A] [answered]
It looks like all the requests are initiated in the user's browser. We wanted to run Hoppscotch in a special perimeter which has access to different endpoints which are not directly accessible by the user's browser. Is there a way to let the backend execute the requests instead of the browser?

**Comment by liyasthomas** [comment, marked as answer]
Hi @OneCyrus, Hoppscotch supports a [Proxy mode](https://docs.hoppscotch.io/documentation/features/interceptor#proxy) which helps you to generate the request from a server.

The official proxy server is hosted by Hoppscotch at:
```
https://proxy.hoppscotch.io
```

You can either use the official public proxy or self-host your own and provide the URL in [Settings page](https://hoppscotch.io/settings) > Interceptor > Proxy.

Source code for Hoppscotch Proxy server is open source: https://github.com/hoppscotch/proxyscotch

**Comment by singhal0306** [comment]
The proxy setting is still at the client side, and hoppscoth backend dont know anything about the proxy as they both are separate entity.
Also i have some internal API, if i host them i want my backend to log those request what's being hit and who is hitting them.
There is no way i can host proxy on my production for such purposes.
